src/interfaces/python/swig_typemaps.i

```diff
@@ -1292,16 +1292,72 @@ TYPEMAP_SPARSEFEATURES_OUT(PyObject,      NPY_OBJECT)
     PyErr_SetString(PyExc_SystemError, $1.what());
     SWIG_fail;
 }
-%rename(_kernel) kernel;
+
+%feature("nothread") _rename_python_function;
+
+%typemap(out) void _rename_python_function "$result = PyErr_Occurred() ? NULL : SWIG_Py_Void();"
+%inline %{
+	static void _rename_python_function(PyObject *type, PyObject *old_name, PyObject *new_name) {
+
+#if PY_VERSION_HEX>=0x03000000
+		if (!PyUnicode_Check(old_name) || !PyUnicode_Check(new_name))
+#else
+		if (!PyString_Check(old_name) || !PyString_Check(new_name))
+#endif
+			{
+				PyErr_SetString(PyExc_TypeError, "'old_name' and 'new_name' need to be strings");
+				return;
+			}
+
+		if (PyType_Check(type)) {
+			PyTypeObject *pytype = (PyTypeObject *)type;
+			PyObject* obj = PyDict_GetItem(pytype->tp_dict, old_name);
+			if (obj == NULL) {
+				PyErr_SetString(PyExc_ValueError, "object definition does not exist in the given type!");
+				return;
+			}
+			PyDict_SetItem(pytype->tp_dict, new_name, obj);
+			PyDict_DelItem(pytype->tp_dict, old_name);
+		}
+
+	  	else if ( PyModule_Check(type)) {
+		  	PyObject* module = PyModule_GetDict(type);
+		 	PyObject* obj = PyDict_GetItem(module, old_name);
+			if (obj == NULL) {
+				PyErr_SetString(PyExc_ValueError, "object definition does not exist in the given module!");
+				return;
+			}
+			PyDict_SetItem(module, new_name, obj);
+			PyDict_DelItem(module, old_name);
+		}
+		else
+			PyErr_SetString(PyExc_ValueError, "'type' is neither a module or a type");
+  }
+%}
+
 
 %pythoncode %{
+import sys
+
+_FACTORIES = [
+		("_distance", "distance"),
```

Comment:
I am being picky here, but what about just having to write this once, e.g. just "distance" and then "_distance" is automatically generated?
Or will there be a case where we want to divert from this pattern?

Reply:
Yes can do that. It was just to give some flexibility, but if we agree to the same pattern can just use `_`.

Reply:
Maybe write it in a way that it makes it easy to customise the hidden name but default is to use the _ prefix